src/libs/onboardingSelectors.ts

```diff
@@ -35,4 +35,19 @@ function hasCompletedHybridAppOnboardingFlowSelector(tryNewDotData: OnyxValue<ty
     return completedHybridAppOnboarding;
 }
 
-export {hasCompletedGuidedSetupFlowSelector, hasCompletedHybridAppOnboardingFlowSelector};
+/**
+ * Selector to get the value of selfTourViewed from the Onyx store
+ *
+ * `undefined` means the value is not loaded yet
+ * `true` means the user has completed the NewDot onboarding flow
+ * `false` means the user has not completed the NewDot onboarding flow
+ */
+function hasSeenTourSelector(onboarding: OnyxValue<typeof ONYXKEYS.NVP_ONBOARDING>): boolean | undefined {
+    if (Array.isArray(onboarding)) {
+        return true;
+    }
```

Comment:
Given [this answer](https://expensify.slack.com/archives/C07HPDRELLD/p1730505334480879?thread_ts=1730161036.738369&cid=C07HPDRELLD), I think that if the onboarding NVP is not set yet, so old account, we still want to show them the tour option. 
```suggestion
    if (Array.isArray(onboarding)) {
        return false;
    }
```